common/core/web/keyboard-processor/src/keyboards/keyboard.ts

```diff
@@ -246,6 +246,12 @@ namespace com.keyman.keyboards {
       return true;
     }
 
+    get usesSupplementaryPlaneChars(): boolean {
+      let kbd = this.scriptObject;
+      // I3319 - SMP extension, I3363 (Build 301)
+      return kbd && ((kbd['KS'] && kbd['KS'] == 1) || kbd['KN'] == 'Hieroglyphic');
+    }
+
```

Comment:
This is used within the first two commits, where SMP handling was only permanently-enabled when in embedded mode.

That said... it's also good to preserve the original logic somewhere in case a Web-Core consumer needs to know this about a keyboard's ruleset for some reason.